src/pages/SixCoinPuzzle.js

```diff
@@ -0,0 +1,131 @@
+import "../styles.css";
+import React, { useState } from "react";
+import toast, { Toaster } from "react-hot-toast";
+import ArrowKeysReact from "arrow-keys-react";
+import Button from "../components/Button";
+import Coin from "../components/Coin";
+
+const SixCoinPuzzle = () => {
+  const numberOfCoinSlots = 7;
+  const startingArray = [1, 1, 1, 0, 2, 2, 2];
+  const winningArray = [2, 2, 2, 0, 1, 1, 1];
+  const [coinTypeArray, setCoinTypeArray] = useState(startingArray);
+  const [selectedCoin, setSelectedCoin] = useState(0);
+  const [moveCounter, setMoveCounter] = useState(0);
+  const [coinSelectText, setCoinSelectText] = useState(
+    "Click to select a coin"
+  );
+
+  const swapCoins = (array, index1, index2) => {
+    [array[index1], array[index2]] = [array[index2], array[index1]];
+    setCoinTypeArray(array);
+    selectCoin(index2);
+    setMoveCounter(moveCounter + 1);
```

Comment:
Because you're only incrementing this value a single time, this is ok. But it's worth knowing that if you try to update state multiple times in a single function, you should pass an updater function, otherwise the state you're trying to update will be stale. https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state